components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpointTest.java

```diff
@@ -221,36 +328,238 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
+
+        requestParams.put(OAuthConstants.SESSION_DATA_KEY_CONSENT, new String[]{sessionDataKeyConsent});
+        requestParams.put(FrameworkConstants.RequestParams.TO_COMMONAUTH, new String[]{"false"});
+        requestParams.put(OAuthConstants.OAuth20Params.SCOPE, new String[]{"openid"});
+        requestParams.put(OAuthConstants.Prompt.CONSENT, new String[]{consent});
+
+        requestAttributes.put(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus.INCOMPLETE);
+
+        mockHttpRequest(requestParams, requestAttributes);
+
+        OAuth2Parameters oAuth2Params = setOAuth2Parameters(scopes, appName, RESPONSE_MODE_FORM_POST, redirectUrl);
+
+        when(consentCacheEntry.getoAuth2Parameters()).thenReturn(oAuth2Params);
+        when(consentCacheEntry.getLoggedInUser()).thenReturn(new AuthenticatedUser());
+
+        mockStatic(OpenIDConnectUserRPStore.class);
+        when(OpenIDConnectUserRPStore.getInstance()).thenReturn(openIDConnectUserRPStore);
+        doNothing().when(openIDConnectUserRPStore).putUserRPToStore(any(AuthenticatedUser.class),
+                anyString(), anyBoolean(), anyString());
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(tokenPersistenceProcessor);
+        when(tokenPersistenceProcessor.getProcessedClientId(anyString())).thenAnswer(new Answer<Object>(){
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                return (String) invocation.getArguments()[0];
+            }
+        });
+
+        mockStatic(IdentityDatabaseUtil.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+        mockStatic(OAuth2Util.OAuthURL.class);
+        when(OAuth2Util.OAuthURL.getOAuth2ErrorPageUrl()).thenReturn(ERROR_PAGE_URL);
+
+        mockEndpointUtil();
+
+        Response response = oAuth2AuthzEndpoint.authorize(httpServletRequest, httpServletResponse);
+        if (response != null) {
+            assertEquals(response.getStatus(), expectedStatus);
+
+            MultivaluedMap<String, Object> responseMetadata = response.getMetadata();
+            assertNotNull(responseMetadata);
+
+            if (expectedError != null) {
+                assertTrue(CollectionUtils.isNotEmpty(responseMetadata.get(HTTPConstants.HEADER_LOCATION)));
+                String location = (String) responseMetadata.get(HTTPConstants.HEADER_LOCATION).get(0);
+                assertTrue(location.contains(expectedError));
+            }
+        }
+    }
+
+    private void mockHttpRequest(final Map<String, String[]> requestParams,
+                                 final Map<String, Object> requestAttributes) {
         doAnswer(new Answer<Object>(){
             @Override
             public Object answer(InvocationOnMock invocation) throws Throwable {
```

Comment:
Lets use do call real

Reply:
httpRequest is a mocked object. Therefore request attributes and parameters are built within the test and passed through mocking. cannot use the real method call